### PR TITLE
Describe NVA Customer API as Gherkin Scenarios

### PR DESCRIPTION
**Pull request by janbj**

### CODE REVIEW

File: customer-commons/src/test/resources/features/get_all_customers.feature [hunk 1]
```diff
@@ -0,0 +1,40 @@
+Feature: Get all Customers
+
+  Scenario: The User requests all Customers
```
Comment (brinxmat) [inline customer-commons/src/test/resources/features/get_all_customers.feature:3]: They should have credentials

File: customer-commons/src/test/resources/features/get_all_customers.feature [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    Given that the User is looking for all Customers
+    When they set the Accept header to "application/json"
+    And they request GET /customer
+    Then they receive a response with status code 200
+    And they see that the response Content-Type header is "application/json"
+    And they see that the response body is a JSON object with a list of Customers
```
Comment (brinxmat) [inline customer-commons/src/test/resources/features/get_all_customers.feature:9]: "JSON array containing customer objects", and I think the other feature for customer covers the checks here

File: customer-commons/src/test/resources/features/get_customer.feature [hunk 3]
```diff
@@ -0,0 +1,63 @@
+Feature: Get Customer
+
+  Scenario Outline: The User requests a Customer
+    Given that the User has a valid <Customer UUID>
```
Comment (brinxmat) [inline customer-commons/src/test/resources/features/get_customer.feature:4]: Credentials

File: customer-commons/src/test/resources/features/get_customer.feature [hunk 4]
```diff
@@ -0,0 +1,63 @@
+Feature: Get Customer
+
```
Comment (brinxmat) [inline customer-commons/src/test/resources/features/get_customer.feature:2]: You can apply a background definition

c.f. https://cucumber.io/docs/gherkin/reference/#background

File: customer-commons/src/test/resources/features/problem/bad_gateway_template.json [hunk 5]
```diff
@@ -0,0 +1,5 @@
+{
```
Comment (brinxmat) [inline customer-commons/src/test/resources/features/problem/bad_gateway_template.json:1]: I was going to use these template files for the tests, so they're maybe superfluous.

File: customer-commons/src/test/resources/features/update_customer.feature [hunk 6]
```diff
@@ -0,0 +1,25 @@
+Feature: Update an existing Customer
+
```
Comment (brinxmat) [inline customer-commons/src/test/resources/features/update_customer.feature:2]: Again, background would make this less DRY-incompatible :)

File: customer-commons/src/test/resources/features/update_customer.feature [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+    And they set the Accept header to "application/json"
+    And they set the Authentication header to "Bearer <credentials>"
+    And they set the request body to a JSON object describing the Customer
+    And they request PUT /customer/<Customer UUUID>
+    Then they receive a response with status code 200
+    And they see that the response Content-type is "application/json"
+    And they see the response body contains a JSON object describing the Customer
+
+  Scenario:
+    Given the user wants to create a new Customer and has valid <Customer UUID> but has invalid <credentials>
```
Comment (brinxmat) [inline customer-commons/src/test/resources/features/update_customer.feature:15]: create?

Comment (brinxmat) [reply customer-commons/src/test/resources/features/update_customer.feature:15]: In fact, I don't think the given is necessary

Comment (janbj) [reply customer-commons/src/test/resources/features/update_customer.feature]: But every Scenario should have a Given?

File: customer-commons/src/test/resources/features/update_customer.feature (hunk 6 above)
Comment (janbj) [reply customer-commons/src/test/resources/features/update_customer.feature:2]: How much would you push into Background? I am now using Administrator and User to show that user has valid credentials or not.

File: customer-commons/src/test/resources/features/update_customer.feature (hunk 7 above)
Comment (brinxmat) [reply customer-commons/src/test/resources/features/update_customer.feature]: Given represents preconditions, if there aren't any I would remove them

File: customer-commons/src/test/resources/features/update_customer.feature (hunk 6 above)
Comment (brinxmat) [reply customer-commons/src/test/resources/features/update_customer.feature:2]: Anything that is repeated in multiple features and not directly related to the state of the particular scenario can be put in the Background.

**Pull request merged**